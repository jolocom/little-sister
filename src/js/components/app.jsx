import React from 'react'
import Reflux from 'reflux'
import Radium from 'radium'
import {bankUri} from 'lib/fixtures'

import {Layout, Content} from 'components/layout'
import {
  Paper,
  AppBar,
  IconButton,
  Snackbar,
  FlatButton,
  Dialog} from 'material-ui'
import Badge from 'material-ui/Badge'
import NavigationMenu from 'material-ui/svg-icons/navigation/menu'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

import JolocomTheme from 'styles/jolocom-theme'

import LeftNav from 'components/left-nav/nav.jsx'
import Profile from 'components/accounts/profile.jsx'
import Tour from 'components/tour.jsx'

import Loading from 'components/common/loading.jsx'

import AccountActions from 'actions/account'
import AccountStore from 'stores/account'
import ConfirmStore from 'stores/confirm'

import PinnedActions from 'actions/pinned'
import ConfirmActions from 'actions/confirm'
import ProfileActions from 'actions/profile'
import ProfileStore from 'stores/profile'

import SnackbarStore from 'stores/snackbar'

// A pathname is considered public if either "/" or if it starts
// with any of the following publicRoutes
const publicRoutes = [
  '/login',
  '/signup',
  '/forgot-password',
  '/change-password'
]

let App = React.createClass({

  mixins: [
    Reflux.connect(AccountStore, 'account'),
    Reflux.connect(ProfileStore, 'profile'),
    Reflux.connect(SnackbarStore, 'snackbar'),
    Reflux.connect(ConfirmStore, 'confirm')
  ],

  propTypes: {
    location: React.PropTypes.object,
    children: React.PropTypes.node,
    route: React.PropTypes.object
  },

  contextTypes: {
    router: React.PropTypes.object.isRequired
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object,
    profile: React.PropTypes.any,
    account: React.PropTypes.object,
    username: React.PropTypes.string,
    searchActive: React.PropTypes.bool,
    location: React.PropTypes.object,
    route: React.PropTypes.object
  },

  getChildContext: function () {
    let {account, profile, searchActive} = this.state
    return {
      muiTheme: this.theme,
      profile,
      account,
      username: account && account.username, // backward compat
      searchActive,
      location: this.props.location,
      route: this.props.route
    }
  },

  getInitialState() {
    return {
      searchActive: false,
      searchQuery: null,
      showIndicator: false
    }
  },

  componentWillMount() {
    this.theme = getMuiTheme(JolocomTheme)
    AccountActions.login()
  },

  componentDidUpdate(prevProps, prevState) {
    let {username} = this.state.account

    if (prevState.account.username === undefined ||
      prevState.account.username !== username) {
      this.checkLogin()
    }
  },

  isPublicRoute(path = this.props.location.pathname) {
    return path === '/' ||
      publicRoutes.some((publicRoute) => path.indexOf(publicRoute) === 0)
  },

  checkLogin() {
    let {username, loggingIn} = this.state.account

    // session is still loading, so return for now
    if (username === undefined && loggingIn) {
      return
    }

    if (!username && !this.isPublicRoute()) {
      this.context.router.push('/login')
    } else if (username && this.isPublicRoute()) {
      this.context.router.push('/graph')
    }

    if (username) {
      ProfileActions.load()
    }
  },

  _handlePinnedTap() {
    PinnedActions.show()
  },

  _handleSearchTap() {
    this.refs.search.show()
    this.setState({searchActive: true})
  },

  _handleSearchChange(query) {
    this.setState({searchQuery: query})
  },

  _handleSearchSubmit() {
    let uri = `${bankUri}/${this.state.searchQuery}#this`
    this.context.router.push(`/graph/${encodeURIComponent(uri)}`)
    this.refs.search.hide()
  },

  _handleSearchHide() {
    this.setState({
      searchActive: false,
      searchQuery: null
    })
  },

  _handleChatTap() {
    this.context.router.push('/conversations')
  },

  showDrawer() {
    this.refs.leftNav.show()
  },

  _handleConfirmClose() {
    ConfirmActions.close()
    this.state.confirm.callback()
  },

  getStyles() {
    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative'
      },
      header: {
        zIndex: 5,
        backgroundColor: this.theme.appBar.color
      },
      bar: {
        boxShadow: 'none'
      },
      nav: {
        display: this.state.searchActive ? 'none' : 'block'
      },
      icon: {
        color: this.theme.appBar.textColor
      },
      filters: {
        width: '100%',
        height: '48px'
      },
      menuIcon: {
        marginTop: '-20px',
        position: 'relative',
        top: '-4px',
        cursor: 'pointer'
      },
      hamburgerBadge: {
        top: 10,
        right: 20,
        width: 15,
        height: 15,
        display: 'none'
      }
    }
    return styles
  },

  render() {
    const styles = this.getStyles()

    // @TODO render login screen when logging in, also makes sures child
    // components don't get rendered before any user data is available
    if (this.state.account.loggingIn && !this.isPublicRoute()) {
      return <Loading />
    }

    // Deactivating search until we get it working
    /* <IconButton
       iconClassName="material-icons"
       iconStyle={styles.icon}
       onTouchTap={this._handleSearchTap}>search</IconButton> */
    const nav = (
      <div>
        <IconButton
          iconClassName="material-icons"
          iconStyle={styles.icon}
          onTouchTap={this._handleChatTap}>chat</IconButton>
      </div>
    )

    // Deactivating search until we get it working
    /*
    (
      <GraphSearch
        ref="search"
        onChange={this._handleSearchChange}
        onSubmit={this._handleSearchSubmit}
        onHide={this._handleSearchHide} />
    )
    */
    const search = null

    // Deactivating the filters until we get them working
    // <GraphFilters style={styles.filters} showDefaults />
    const filters = null

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={this._handleConfirmClose}
      />,
      <FlatButton
        label={this.state.confirm.primaryActionText}
        primary
        onTouchTap={this._handleConfirmClose}
      />
    ]

    return (
      <div style={styles.container}>
        {this.isPublicRoute() ? this.props.children : (
          <Layout>
            <Paper style={styles.header}>
              <AppBar
                title="Graph"
                iconElementRight={nav}
                style={styles.bar}
                iconElementLeft={
                  <Badge
                    badgeContent={''}
                    secondary
                    badgeStyle={styles.hamburgerBadge}>
                    <NavigationMenu
                      onTouchTap={this.showDrawer}
                      style={styles.menuIcon} />
                  </Badge>
                }
                onLeftIconButtonTouchTap={this.showDrawer} />
              {filters}
              {search}
            </Paper>
            <LeftNav ref="leftNav" />
            <Content>
              {React.Children.map(this.props.children, (el) => {
                return React.cloneElement(el, {
                  searchQuery: this.state.searchQuery
                })
              })}
            </Content>
            <Profile />
            <Tour />
          </Layout>
        )}

        <Snackbar
          open={this.state.snackbar.open}
          message={this.state.snackbar.message}
          action={this.state.snackbar.undo && 'undo'}
          onActionTouchTap={this.state.snackbar.undoCallback}
        />

        <Dialog
          actions={confirmActions}
          modal={false}
          open={this.state.confirm.open}
          onRequestClose={this.handleClose}
        >
          {this.state.confirm.message}
        </Dialog>
      </div>

    )
  }

})

export default Radium(App)
