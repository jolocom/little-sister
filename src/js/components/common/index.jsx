export { default as IconIdCard } from './icon-idcard'
export { default as IconPassport } from './icon-passport'
export { default as IconServices } from './icon-services'
export { default as Loading } from './loading'
export { default as Error } from './error'
export { default as Spinner } from './spinner'
export { default as IconCheckmark } from './icon-checkmark'
export { default as IconEther } from './icon-ether'
export { default as IconBlockchain } from './icon-blockchain'
