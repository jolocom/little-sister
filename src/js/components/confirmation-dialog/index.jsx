import React from 'react'
import { connect } from 'redux/utils'

import { FlatButton, Dialog } from 'material-ui'

@connect({
  props: ['confirm'],
  actions: ['confirmation-dialog:close']
})
export default class ConfirmationDialog extends React.Component {
  static propTypes = {
    close: React.PropTypes.func,
    confirm: React.PropTypes.object
  }

  _handleConfirmAction() {
    this.props.close()
    this.props.confirm.callback() // Action when the user confirms
  }

  _handleConfirmCancel() {
    this.props.close()
  }

  render() {
    const confirmHandler = () => this._handleConfirmAction()
    const cancelHandler = () => this._handleConfirmCancel()

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={cancelHandler}
      />,
      <FlatButton
        label={this.props.confirm.primaryActionText}
        primary
        onTouchTap={confirmHandler}
      />
    ]

    return <Dialog
      actions={confirmActions}
      modal={false}
      open={this.props.confirm.open}
      onRequestClose={this.handleClose}
    >
      {this.props.confirm.message}
    </Dialog>
  }
}
