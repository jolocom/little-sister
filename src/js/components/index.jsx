import React from 'react'
import Radium from 'radium'
import {RaisedButton} from 'material-ui'
import Carousel from 're-carousel'
import IndicatorDots from 'components/common/indicator-dots.jsx'

let Index = React.createClass({
  contextTypes: {
    router: React.PropTypes.object,
    username: React.PropTypes.string,
    muiTheme: React.PropTypes.object
  },

  componentWillMount() {
    const {account} = this.context
    if (account && account.webId) {
      this.context.router.push('/graph')
    }
  },

  getStyles() {
    let {muiTheme} = this.context

    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        backgroundColor: muiTheme.jolocom.gray4
      },
      logo: {
        fontSize: '18px',
        fontWeight: '400',
        textAlign: 'center',
        marginTop: '24px',
        textTransform: 'uppercase'
      },
      logoImg: {
        width: '32px',
        height: '32px',
        verticalAlign: 'middle'
      },
      intro: {
        flex: 1,
        textAlign: 'center'
      },
      slide: {
        backgroundColor: muiTheme.jolocom.gray4,
        padding: '24px',
        height: '100%',
        display: 'flex',
        flexDirection: 'column'
      },
      content: {
        flex: 1
      },
      img: {
        maxWidth: '100%'
      },
      title: {
        color: muiTheme.jolocom.gray1,
        fontSize: '28px',
        fontWeight: '300'
      },
      logoStartImg: {
        padding: '24px',
        maxWidth: '80%',
        width: '320px',
        margin: 'auto',
        marginTop: '20vh'
      },
      actions: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'stretch',
        padding: '24px',
        maxWidth: '80%',
        width: '260px',
        margin: 'auto'
      },
      signup: {
        marginBottom: '12px'
      },
      login: {
      }
    }

    return styles
  },

  render() {
    let styles = this.getStyles()

    return (
      <div style={styles.container}>

        {/** <div style={styles.logo}>
          <img src="/img/logo.png" style={styles.logoImg} /> Jolocom
        </div> **/}

        {/** <Carousel style={styles.intro} indicator={IndicatorDots}>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-data.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Create your personal digital identity</h3>
            <div style={styles.content}>
              <img src="/img/slide-identity.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-share.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Visualize your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-visualize.png" style={styles.img} />
            </div>
          </div>
        </Carousel> **/}

        <div style={styles.intro}>
          <img
            src="/img/logo_littlesister_start.svg"
            style={styles.logoStartImg} />
        </div>

        <div style={styles.actions}>
          <RaisedButton
            secondary
            label="Sign up"
            style={styles.signup}
            onClick={this._handleSignup}
          />
          <RaisedButton
            label="Log in"
            style={styles.login}
            onClick={this._handleLogin}
          />
        </div>
      </div>
    )
  },

  _handleSignup() {
    this.context.router.push('/signup')
  },

  _handleLogin() {
    this.context.router.push('/login')
  }

})

export default Radium(Index)
