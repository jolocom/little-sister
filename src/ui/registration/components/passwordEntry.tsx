import React from 'react'
import { StyleSheet, TextInput } from 'react-native'
import { Button } from 'react-native-material-ui'
import { Container, Block, CenteredText } from 'src/ui/structure'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import I18n from 'src/locales/i18n'

interface Props {
  lineColorPassword: string
  lineColorRepeat: string
  keyboardDrawn: boolean
  password: string
  confirmPassword: string
  onPasswordChange: (input: string) => void
  onPasswordConfirmChange: (input: string) => void
  clickNext: () => void
  onFocusRepeat: () => void
  onBlurRepeat: () => void
  onFocusPassword: () => void
  onBlurPassword: () => void
}

// TODO REFACTOR, reference stylesheet for fonts, no magic numbers
// Use flex more

const styles = StyleSheet.create({
  subHeader: {
    fontSize: 20,
    fontFamily: JolocomTheme.contentFontFamily,
    fontWeight: '100',
    padding: '10%',
    color: JolocomTheme.primaryColorSand,
  },
  infoPassword: {
    fontSize: 14,
    fontWeight: '100',
    fontFamily: JolocomTheme.contentFontFamily,
    opacity: 0.6,
    padding: '10%',
    color: JolocomTheme.primaryColorSand,
  },
  textInputField: {
    padding: '3%',
    color: JolocomTheme.primaryColorWhite,
    fontSize: 22,
    fontWeight: '100',
    fontFamily: JolocomTheme.contentFontFamily,
    width: '80%',
  },
  textErrorField: {
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorSandInactive,
    fontWeight: '100',
    fontSize: 14,
  },
  mainContainer: {
    justifyContent: 'space-between',
    backgroundColor: JolocomTheme.primaryColorBlack,
  },
  nestedContainer: {
    justifyContent: 'space-around',
  },
  buttonContainer: {
    width: 164,
    height: 48,
    borderRadius: 4,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorWhite,
    fontSize: JolocomTheme.headerFontSize,
    fontWeight: '100',
  },
  buttonTextDisabled: {
    paddingVertical: 15,
    fontFamily: JolocomTheme.contentFontFamily,
    color: 'rgba(255,255,255, 0.4)',
    fontSize: JolocomTheme.headerFontSize,
    fontWeight: '100',
  },
})

export const PasswordEntryComponent: React.SFC<Props> = props => {
  const { password, confirmPassword, keyboardDrawn } = props
  const errorMsg = validateInput(password, confirmPassword)

  return (
    <Container style={styles.mainContainer}>
      <Block style={styles.nestedContainer} flex={0.4}>
        <CenteredText
          style={styles.subHeader}
          msg={
            keyboardDrawn
              ? ''
              : I18n.t('Set a password to encrypt your data on the device')
          }
        />
        <CenteredText
          style={styles.infoPassword}
          msg={
            I18n.t('This password will be stored in your keychain') +
            '. ' +
            I18n.t(
              'After setting it, please make sure you have passcode enabled',
            ) +
            '.'
          }
        />
      </Block>
      <Block style={styles.nestedContainer} flex={0.3}>
        <TextInput
          style={styles.textInputField}
          placeholder={I18n.t('Password')}
          maxLength={40}
          editable
          secureTextEntry
          keyboardType={'default'}
          onChangeText={props.onPasswordChange}
          onFocus={props.onFocusPassword}
          onBlur={props.onBlurPassword}
          placeholderTextColor={JolocomTheme.primaryColorSandInactive}
          selectionColor={JolocomTheme.primaryColorPurple}
          underlineColorAndroid={props.lineColorPassword}
        />
        <TextInput
          style={styles.textInputField}
          placeholder={I18n.t('Repeat password')}
          maxLength={40}
          editable
          secureTextEntry
          keyboardType={'default'}
          onChangeText={props.onPasswordConfirmChange}
          onFocus={props.onFocusRepeat}
          onBlur={props.onBlurRepeat}
          placeholderTextColor={JolocomTheme.primaryColorSandInactive}
          selectionColor={JolocomTheme.primaryColorPurple}
          underlineColorAndroid={props.lineColorRepeat}
        />
        <CenteredText
          style={styles.textErrorField}
          msg={password.length > 5 ? errorMsg : ''}
        />
      </Block>
      <Block flex={0.1}>
        <Button
          style={
            !errorMsg
              ? { container: styles.buttonContainer, text: styles.buttonText }
              : {
                  container: styles.buttonContainer,
                  text: styles.buttonTextDisabled,
                }
          }
          onPress={props.clickNext}
          raised
          upperCase={false}
          disabled={!!errorMsg}
          text={I18n.t('Continue')}
        />
      </Block>
    </Container>
  )
}

const validateInput = (password: string, confirmPassword: string): string => {
  if (password.indexOf(' ') !== -1) {
    return I18n.t('No spaces allowed')
  }

  if (!password.match(/[A-Z]/)) {
    return I18n.t('At least one uppercase letter needed')
  }

  if (!password.match(/[0-9]/)) {
    return I18n.t('At least one number needed')
  }

  if (password !== confirmPassword) {
    return I18n.t('Passwords do not match')
  }

  if (password.length < 8) {
    return I18n.t('At least 8 characters are required')
  }

  return ''
}
