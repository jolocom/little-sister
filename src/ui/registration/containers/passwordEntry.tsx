import React from 'react'
import { connect } from 'react-redux'
import { Keyboard, EmitterSubscription } from 'react-native'
import { registrationActions } from 'src/actions'
import { PasswordEntryComponent } from 'src/ui/registration/components/passwordEntry'
import { JolocomTheme } from 'src/styles/jolocom-theme'

interface ConnectProps {
  savePassword: (password: string) => void
}

interface Props extends ConnectProps {}

interface State {
  password: string
  confirmPassword: string
  keyboardDrawn: boolean
  lineColorPassword: string
  lineColorRepeat: string
}

export class PasswordEntryContainer extends React.Component<Props, State> {
  private kbShowListener!: EmitterSubscription
  private kbHideListener!: EmitterSubscription

  state = {
    password: '',
    confirmPassword: '',
    keyboardDrawn: false,
    lineColorPassword: JolocomTheme.primaryColorSand,
    lineColorRepeat: JolocomTheme.primaryColorSand,
  }

  componentDidMount() {
    this.setupListeners()
  }
  componentWillUnmount() {
    this.removeListeners()
  }

  private setupListeners(): void {
    this.kbShowListener = Keyboard.addListener('keyboardDidShow', () =>
      this.setState({ keyboardDrawn: true }),
    )

    this.kbHideListener = Keyboard.addListener('keyboardDidHide', () =>
      this.setState({ keyboardDrawn: false }),
    )
  }

  private removeListeners(): void {
    this.kbShowListener.remove()
    this.kbHideListener.remove()
  }

  private handleOnConfirm = (): void => {
    Keyboard.dismiss()
    this.props.savePassword(this.state.password)
  }

  private onPasswordChange = (password: string): void => {
    this.setState({ password })
  }

  private onFocusRepeat = (): void => {
    this.setState({
      lineColorRepeat: JolocomTheme.primaryColorPurple,
    })
  }

  private onFocusPassword = (): void => {
    this.setState({
      lineColorPassword: JolocomTheme.primaryColorPurple,
    })
  }

  private onBlurRepeat = (): void => {
    this.setState({
      lineColorRepeat: JolocomTheme.primaryColorSand,
    })
  }

  private onBlurPassword = (): void => {
    this.setState({
      lineColorPassword: JolocomTheme.primaryColorSand,
    })
  }

  private onPasswordConfirmChange = (confirmPassword: string): void => {
    this.setState({ confirmPassword })
  }

  render() {
    return (
      <PasswordEntryComponent
        onPasswordChange={this.onPasswordChange}
        onPasswordConfirmChange={this.onPasswordConfirmChange}
        password={this.state.password}
        confirmPassword={this.state.confirmPassword}
        keyboardDrawn={this.state.keyboardDrawn}
        clickNext={this.handleOnConfirm}
        onFocusRepeat={this.onFocusRepeat}
        onBlurRepeat={this.onBlurRepeat}
        onFocusPassword={this.onFocusPassword}
        onBlurPassword={this.onBlurPassword}
        lineColorPassword={this.state.lineColorPassword}
        lineColorRepeat={this.state.lineColorRepeat}
      />
    )
  }
}

const mapDispatchToProps = (dispatch: Function) => ({
  savePassword: (password: string) => {
    dispatch(registrationActions.savePassword(password))
  },
})

export const PasswordEntry = connect(
  null,
  mapDispatchToProps,
)(PasswordEntryContainer)
