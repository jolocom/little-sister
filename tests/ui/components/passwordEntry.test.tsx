import React from 'react'
import { PasswordEntryComponent } from 'src/ui/registration/components/passwordEntry'
import { shallow } from 'enzyme'

describe('passwordEntry component', ()=> {
  const COMMON_PROPS = {
    onPasswordChange: () => null,
    onPasswordConfirmChange: () => null,
    clickNext: () => null
  }

  it('matches the snapshot when no input is present', () => {
    const props = Object.assign({}, COMMON_PROPS {
      keyboardDrawn: false,
      password: '',
      confirmPassword: ''
    })

    const rendered = shallow(<PasswordEntryComponent {...props}/>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot when keyboard is drawn', () => {
    const props = Object.assign({}, COMMON_PROPS {
      keyboardDrawn: true,
      password: '',
      confirmPassword: ''
    })

    const rendered = shallow(<PasswordEntryComponent {...props}/>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot when passwords are not valid', () => {
    const props = Object.assign({}, COMMON_PROPS {
      keyboardDrawn: true,
      password: 'nodigitanduppercase',
      confirmPassword: ''
    })

    const rendered = shallow(<PasswordEntryComponent {...props}/>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot when passwords are valid', () => {
    const props = Object.assign({}, COMMON_PROPS, {
      keyboardDrawn: true,
      password: 'ValidPass1',
      confirmPassword: 'ValidPass1',
    }

    const rendered = shallow(<PasswordEntryComponent {...props}/>)
    expect(rendered).toMatchSnapshot()
  })
})
