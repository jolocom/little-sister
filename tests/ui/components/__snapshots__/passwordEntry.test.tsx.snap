// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwordEntry component matches the snapshot when keyboard is drawn 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
      "justifyContent": "space-between",
    }
  }
>
  <Block
    flex={0.4}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <Header
      title=""
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 20,
          "padding": "10%",
        }
      }
    />
    <CenteredText
      msg="This password will be stored in your keychain. After setting it, please make sure you have passcode enabled."
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 14,
          "opacity": 0.6,
          "padding": "10%",
        }
      }
    />
  </Block>
  <Block
    flex={0.3}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Repeat password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "rgba(255, 239, 223, 0.38)",
          "fontSize": 14,
        }
      }
    />
  </Block>
  <Block
    flex={0.1}
  >
    <Button
      accent={false}
      disabled={true}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={true}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "height": 48,
            "width": 164,
          },
          "text": Object {
            "backgroundColor": "#942f51",
            "fontFamily": "TTCommons",
            "fontSize": 22,
            "fontWeight": "100",
          },
        }
      }
      testID={null}
      text="Continue"
      upperCase={false}
    />
  </Block>
</Container>
`;

exports[`passwordEntry component matches the snapshot when no input is present 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
      "justifyContent": "space-between",
    }
  }
>
  <Block
    flex={0.4}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <Header
      title="Step 2"
    />
    <CenteredText
      msg="Set a password to encrypt your data on the device"
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 20,
          "padding": "10%",
        }
      }
    />
    <CenteredText
      msg="This password will be stored in your keychain. After setting it, please make sure you have passcode enabled."
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 14,
          "opacity": 0.6,
          "padding": "10%",
        }
      }
    />
  </Block>
  <Block
    flex={0.3}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Repeat password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "rgba(255, 239, 223, 0.38)",
          "fontSize": 14,
        }
      }
    />
  </Block>
  <Block
    flex={0.1}
  >
    <Button
      accent={false}
      disabled={true}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={true}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "height": 48,
            "width": 164,
          },
          "text": Object {
            "backgroundColor": "#942f51",
            "fontFamily": "TTCommons",
            "fontSize": 22,
            "fontWeight": "100",
          },
        }
      }
      testID={null}
      text="Continue"
      upperCase={false}
    />
  </Block>
</Container>
`;

exports[`passwordEntry component matches the snapshot when passwords are not valid 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
      "justifyContent": "space-between",
    }
  }
>
  <Block
    flex={0.4}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <Header
      title=""
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 20,
          "padding": "10%",
        }
      }
    />
    <CenteredText
      msg="This password will be stored in your keychain. After setting it, please make sure you have passcode enabled."
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 14,
          "opacity": 0.6,
          "padding": "10%",
        }
      }
    />
  </Block>
  <Block
    flex={0.3}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Repeat password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <CenteredText
      msg="At least one uppercase letter needed"
      style={
        Object {
          "color": "rgba(255, 239, 223, 0.38)",
          "fontSize": 14,
        }
      }
    />
  </Block>
  <Block
    flex={0.1}
  >
    <Button
      accent={false}
      disabled={true}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={true}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "height": 48,
            "width": 164,
          },
          "text": Object {
            "backgroundColor": "#942f51",
            "fontFamily": "TTCommons",
            "fontSize": 22,
            "fontWeight": "100",
          },
        }
      }
      testID={null}
      text="Continue"
      upperCase={false}
    />
  </Block>
</Container>
`;

exports[`passwordEntry component matches the snapshot when passwords are valid 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
      "justifyContent": "space-between",
    }
  }
>
  <Block
    flex={0.4}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <Header
      title=""
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 20,
          "padding": "10%",
        }
      }
    />
    <CenteredText
      msg="This password will be stored in your keychain. After setting it, please make sure you have passcode enabled."
      style={
        Object {
          "color": "#ffefdf",
          "fontFamily": "TTCommons",
          "fontSize": 14,
          "opacity": 0.6,
          "padding": "10%",
        }
      }
    />
  </Block>
  <Block
    flex={0.3}
    style={
      Object {
        "justifyContent": "space-around",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Repeat password"
      placeholderTextColor="rgba(255, 239, 223, 0.38)"
      secureTextEntry={true}
      selectionColor="#942f51"
      style={
        Object {
          "color": "white",
          "fontFamily": "TTCommons",
          "fontSize": 22,
          "padding": "3%",
          "width": "80%",
        }
      }
    />
    <CenteredText
      msg=""
      style={
        Object {
          "color": "rgba(255, 239, 223, 0.38)",
          "fontSize": 14,
        }
      }
    />
  </Block>
  <Block
    flex={0.1}
  >
    <Button
      accent={false}
      disabled={false}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={true}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "height": 48,
            "width": 164,
          },
          "text": Object {
            "backgroundColor": "#942f51",
            "fontFamily": "TTCommons",
            "fontSize": 22,
            "fontWeight": "100",
          },
        }
      }
      testID={null}
      text="Continue"
      upperCase={false}
    />
  </Block>
</Container>
`;
